signature/src/policy.rs

```diff
@@ -0,0 +1,893 @@
+}
+
+impl PolicyReqSimple {
+    fn simple_judge_allowed(&self, _image: &image::Image) -> Result<()> {
+        if self.r#type == PolicyReqType::TypeReject.to_string() {
+            return Err(anyhow!(r#"The policy is "reject""#));
+        } else if self.r#type == PolicyReqType::TypeAccept.to_string() {
+            Ok(())
+        } else {
+            Err(anyhow!(ErrorInfo::ErrUnknownPrType.to_string()))
+        }
+    }
+}
+
+// The `signedBy` policy requirement:
+// The image must be signed by trusted keys for a specified identity
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct PolicyReqSignedBy {
+    r#type: String,
+
+    // KeyType specifies what kind of key reference KeyPath/KeyData is.
+    // Acceptable values are "GPGKeys" | "signedByGPGKeys” "X.509Certificates" | "signedByX.509CAs"
+    // FIXME: now only support "GPGKeys", fllowing the [containers/image](https://github.com/containers/image)
+    #[serde(rename = "keyType")]
+    key_type: String,
+
+    // KeyPath is a pathname to a local file containing the trusted key(s).
+    // Exactly one of KeyPath and KeyData must be specified.
+    #[serde(default, rename = "keyPath")]
+    key_path: String,
+    // KeyData contains the trusted key(s), base64-encoded.
+    // Exactly one of KeyPath and KeyData must be specified.
+    #[serde(default, rename = "keyData")]
+    key_data: String,
+
+    // SignedIdentity specifies what image identity the signature must be claiming about the image.
+    // Defaults to "match-exact" if not specified.
+    #[serde(default, rename = "signedIdentity")]
+    signed_identity: Option<PolicyReferenceMatch>,
+}
+
+impl PolicyReqSignedBy {
+    fn signature_judge_allowed(&self, image: &mut image::Image) -> Result<()> {
+        if self.r#type != PolicyReqType::TypeSignedBy.to_string() {
+            return Err(anyhow!(ErrorInfo::ErrUnknownPrType.to_string()));
+        }
+
+        let sigs = image.get_signatures()?;
+        if sigs.is_empty() {
+            return Err(anyhow!("Can not find any signatures."));
+        }
+
+        let mut reject_reason: Vec<anyhow::Error> = Vec::new();
+
+        for sig in sigs.iter() {
+            match self.is_signature_valid(image, sig.to_vec()) {
+                // One accepted signature is enough.
+                Ok(()) => {
+                    return Ok(());
+                }
+                Err(e) => {
+                    reject_reason.push(e);
+                }
+            }
+        }
+
+        Err(anyhow!(format!(
+            "The signatures do not satisfied! Reject reason: {:?}",
+            reject_reason
+        )))
+    }
+
+    #[allow(unused_assignments)]
+    fn is_signature_valid(&self, image: &image::Image, sig: Vec<u8>) -> Result<()> {
+        // FIXME: only support "GPGKeys" type now.
+        if self.key_type != KeyType::Gpg.to_string() {
+            return Err(anyhow!(
+                "UNKNOWN KEY TYPE in policy config: only support 'GPGKeys' now."
```

Comment:
- I would make the entire message lower case.
- Why not add the value of `KeyType::Gpg.to_string()` to the error with a format rather than hard-coding it?